src/main/java/com/cloudbees/jenkins/plugins/bitbucket/server/client/BitbucketServerAPIClient.java

```diff
@@ -108,6 +86,12 @@
     private static final String API_COMMIT_STATUS_PATH = "/rest/build-status/1.0/commits/%s";
 
     private static final int MAX_PAGES = 100;
+    private static final MultiThreadedHttpConnectionManager connectionManager = new MultiThreadedHttpConnectionManager();
+
+    static {
+        connectionManager.getParams().setDefaultMaxConnectionsPerHost(20);
```

Comment:
I would use an static field to store the actual value and also try to read from a system variable first using `20` as a fallback default.

[It is a common pattern in Jenkins](https://github.com/jenkinsci/jenkins/blob/68f55859994c57e9942c5fdffde64dc1a66990b2/core/src/main/java/hudson/model/AsyncAperiodicWork.java#L53).